src/scores/spatial/fss_impl.py

```diff
@@ -0,0 +1,371 @@
+    their own aggregation methods, rather than ``xarray.ufunc``
+
+    .. warning::
+        ``dask`` option is not fully tested
+
+    Args:
+        fcst: An array of forecasts
+        obs: An array of observations (same spatial shape as ``fcst``)
+        event_threshold: A scalar to compare ``fcst`` and ``obs`` fields to generate a
+            binary "event" field.
+        window_size: A pair of positive integers ``(height, width)`` of the sliding
+            window_size; the window size must be greater than 0 and fit within
+            the shape of ``obs`` and ``fcst``.
+        spatial_dims: A pair of dimension names ``(x, y)`` where ``x`` and ``y`` are
+            the spatial dimensions to slide the window along to compute the FSS.
+            e.g. ``("lat", "lon")``.
+        zero_padding: If set to true, applies a 0-valued window border around the
+            data field before computing the FSS. If set to false (default), it
+            uses the edges (thickness = window size) of the input fields as the border.
+            One can think of it as:
+            - zero_padding = False => inner border
+            - zero_padding = True => outer border with 0 values
+        reduce_dims: Optional set of additional dimensions to aggregate over
+            (mutually exclusive to ``preserve_dims``).
+        preserve_dims: Optional set of dimensions to keep (all other dimensions
+            are reduced); By default all dimensions except ``spatial_dims``
+            are preserved (mutually exclusive to ``reduce_dims``).
+        threshold_operator: The threshold operator used to generate the binary
+            event field. E.g. ``np.greater``. Note: this may depend on the backend
+            ``compute method`` and not all operators may be supported. Generally,
+            ``np.greater``, ``np.less`` and their counterparts. Defaults to "np.greater".
+        compute_method: currently only supports :py:obj:`FssComputeMethod.NUMPY`
+            see: :py:class:`scores.fast.fss.typing.FssComputeMethod`
+        dask: See ``xarray.apply_ufunc`` for options
+
+    Returns:
+        An ``xarray.DataArray`` containing the FSS computed over the ``spatial_dims``
+
+        The resultant array will have the score grouped against the remaining
+        dimensions, unless `reduce_dims`/`preserve_dims` are specified; in which
+        case, they will be aggregated over the specified dimensions accordingly.
+
+        For an exact usage please refer to ``FSS.ipynb`` in the tutorials.
+
+    Raises:
+        DimensionError: Various errors are thrown if the input dimensions
+            do not conform. e.g. if the window size is larger than the input
+            arrays, or if the the spatial dimensions in the args are missing in
+            the input arrays.
+
+        DimensionWarning: If ``spatial_dims`` are attempting to be preserved e.g. in ``preserve_dims``
+    References:
```

Comment:
```suggestion

    References:
```

Is it worth adding an empty line before "References" (might it make it a bit easier to read)?

Reply:
Yes.